contracts/strategies/ELKILPStrategyV5.sol

```diff
@@ -0,0 +1,318 @@
+            // deployment checks for non-pool2
+            require(_swapPairToken0 > address(0), "Swap pair 0 is necessary but not supplied");
+            require(_swapPairToken1 > address(0), "Swap pair 1 is necessary but not supplied");
+            swapPairToken0 = IPair(_swapPairToken0);
+            swapPairToken1 = IPair(_swapPairToken1);
+            require(swapPairToken0.token0() == _rewardToken || swapPairToken0.token1() == _rewardToken, "Swap pair supplied does not have the reward token as one of it's pair");
+            require(
+                swapPairToken0.token0() == IPair(address(depositToken)).token0() || swapPairToken0.token1() == IPair(address(depositToken)).token0(),
+                "Swap pair 0 supplied does not match the pair in question"
+            );
+            require(
+                swapPairToken1.token0() == IPair(address(depositToken)).token1() || swapPairToken1.token1() == IPair(address(depositToken)).token1(),
+                "Swap pair 1 supplied does not match the pair in question"
+            );
+        } else if (_rewardToken == IPair(address(depositToken)).token0()) {
+            swapPairToken1 = IPair(address(depositToken));
+        } else if (_rewardToken == IPair(address(depositToken)).token1()) {
+            swapPairToken0 = IPair(address(depositToken));
+        }
+    }
+
+    function setAllowances() public override onlyOwner {
+        depositToken.approve(address(stakingContract), MAX_UINT);
+    }
+
+    function deposit(uint amount) external override {
+        _deposit(msg.sender, amount);
+    }
+
+    function depositWithPermit(uint amount, uint deadline, uint8 v, bytes32 r, bytes32 s) external override {
+        depositToken.permit(msg.sender, address(this), amount, deadline, v, r, s);
+        _deposit(msg.sender, amount);
+    }
+
+    function depositFor(address account, uint amount) external override {
+        _deposit(account, amount);
+    }
+
+    function _deposit(address account, uint amount) private onlyAllowedDeposits {
+        require(DEPOSITS_ENABLED == true, "ELKILPStrategy::_deposit");
+        if (MAX_TOKENS_TO_DEPOSIT_WITHOUT_REINVEST > 0) {
+            uint unclaimedRewards = checkReward();
+            if (unclaimedRewards > MAX_TOKENS_TO_DEPOSIT_WITHOUT_REINVEST) {
+                _reinvest(unclaimedRewards);
+            }
+        }
+
+        require(depositToken.transferFrom(msg.sender, address(this), amount));
+        _stakeDepositTokens(amount);
+        _mint(account, getSharesForDepositTokens(amount));
+        totalDeposits = totalDeposits.add(amount);
+        emit Deposit(account, amount);
+    }
+
+    function withdraw(uint amount) external override {
+        uint coverage = checkCoverage();
+
+        if ( coverage > 0 ) {
+            stakingContract.getCoverage();
+            _reinvest(coverage);
+        }
+
+        uint depositTokenAmount = getDepositTokensForShares(amount); //
```

Comment:
does the coverage+shares cover for the extra elk you were supposed to receive?

Reply:
Not sure I got your point here ?

Reply:
The amount that we get as coverage is reinvested, so it should be accounted for via the `getDepositTokensFOrShares` function no ?

Reply:
at what point is it reinvested? when we call reinvest? or is it done automatically overtime on the elk staking rewards contract?

Reply:
oh it's reinvested automatically if there's coverage, I see, should be grand then

Reply:
Yes in the withdraw function, at the beginning it checks if there is an amount available as coverage, then reinvest it (iif it's greater than 0)